main/test/ca/uwaterloo/flix/library/Core/Io/TestFile.flix

```diff
@@ -117,3 +117,33 @@ def isWritable03(): Bool & Impure = isWritable(new("./doc/foo")) |> Result.isOk
 
 @test
 def testReadLines01(): Bool & Impure = readLines(new("README.md")) |> Result.isOk
+
+@test
+def testCanonical01(): Bool & Impure = canonical(new("./doc/..")) == canonical(new("."))
+
+@test
+def testCanonical02(): Bool & Impure = canonical(new("./doc/foo/..")) == canonical(new("./doc"))
+
+@test
+def testCanonical03(): Bool & Impure = canonical(new("./doc/foo/bar/../bar")) == canonical(new("./doc/foo/bar"))
+
+@test
+def testCombine01(): Bool & Impure = combine(Nel(".", "doc" :: Nil)) == new("./doc/")
+
+@test
+def testCombine02(): Bool & Impure = combine(Nel(".",  "doc" :: ".." :: Nil)) == new("./doc/..")
+
+@test
+def testCombine03(): Bool & Impure = combine(Nel(".", Nil)) == new(".")
+
+@test
+def testAdd01(): Bool & Impure = add(new("./foo"), "bar") == new("./foo/bar")
+
+@test
+def testAdd02(): Bool & Impure = add(new("./foo/.."), "foo") == new("./foo/../foo")
+
+@test
+def testAdd03(): Bool & Impure = add(new("."), "foo") == new("./foo")
+
+// def resEq(a: a, o: Result[a, e]): Bool = Result.getWithDefault(Result.map(b -> a == b, o), false)
```

Comment:
What is this?

Reply:
A long forgotten remnant of a time long gone... or just something I forgot to delete :P